src/flash.rs

```diff
@@ -224,22 +255,47 @@ impl<'a, const SECTOR_SZ_KB: u32> FlashWriter<'a, SECTOR_SZ_KB> {
         // Unlock Flash
         self.unlock()?;
 
-        for idx in (0..data.len()).step_by(2) {
+        for idx in (0..data.len()).step_by(8) {
```

Comment:
IMHO a comment like the following would be nice:
```suggestion
        // According to RM0440 Rev 7, "It is only possible to program double word (2 x 32-bit data)"
        for idx in (0..data.len()).step_by(8) {
```

Reply:
No objections